contrib/cloud.google.com/go/pubsub.v2/internal/tracing/tracing.go

```diff
@@ -0,0 +1,124 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+// Package tracing contains tracing logic for the cloud.google.com/go/pubsub.v2 instrumentation.
+//
+// WARNING: this package SHOULD NOT import cloud.google.com/go/pubsub/v2.
+//
+// The motivation of this package is to support orchestrion, which cannot use the main package because it imports
+// the cloud.google.com/go/pubsub/v2 package, and since orchestrion modifies the library code itself,
+// this would cause an import cycle.
+package tracing
```

Comment:
it seems this package is pretty much the same as the one in v1. Could we reuse it to work for both v1 and v2 under `contrib/cloud.google.com/go/pubsubtrace`? (you can check `contrib/confluentinc/confluent-kafka-go/kafkatrace` as reference where we did something similar)